src/Dapr.Client/TypeConverters.cs

```diff
@@ -21,65 +21,36 @@ internal static class TypeConverters
         /// <param name="options">The JSON serialization options.</param>
         /// <typeparam name="T">The type of the given data.</typeparam>
         /// <returns>The given data as JSON based byte string.</returns>
-        public static ByteString ToJsonByteString<T>(T data, JsonSerializerOptions options = null)
+        public static ByteString ToJsonByteString<T>(T data, JsonSerializerOptions options)
         {
-            if (data == null)
-            {
-                return ByteString.Empty;
-            }
-
             var bytes = JsonSerializer.SerializeToUtf8Bytes(data, options);
             return ByteString.CopyFrom(bytes);
         }
 
-        /// <summary>
-        /// Converts an arbitrary type to the <see cref="Any"/> Protocol Buffer type.
-        ///
-        /// If the given type is a subtype of <see cref="IMessage"/>, then it's save to use the Protocol Buffer
-        /// packaging provided for the <see cref="Any"/> type with the <see cref="Any.Pack(Google.Protobuf.IMessage)"/>.
-        /// For all other types, we use JSON serialization based on <see cref="System.Text.Json"/>.
-        /// </summary>
-        /// <param name="data">The data to convert.</param>
-        /// <param name="options">The JSON serialization options.</param>
-        /// <typeparam name="T">The type of the given data.</typeparam>
-        /// <returns>The <see cref="Any"/> type representation of the given data.</returns>
-        public static Any ToAny<T>(T data, JsonSerializerOptions options = null)
+        public static Any ToJsonAny<T>(T data, JsonSerializerOptions options)
         {
-            if (data == null)
-            {
-                return new Any();
-            }
-
-            var t = typeof(T);
+            return new Any() 
+            { 
+                Value = ToJsonByteString<T>(data, options),
 
-            return typeof(IMessage).IsAssignableFrom(t)
-                ? Any.Pack((IMessage) data)
-                : new Any {Value = ToJsonByteString(data, options), TypeUrl = t.FullName};
```

Comment:
`TypeUrl = t.FullName` was doing something wierd and invalid. The Dapr runtime gives special meaning to "empty" `TypeUrl` even though that's technically invalid also. There's no real reason for us to put the .NET type name here, it has no meaning.